tests/test_mixins.py

```diff
@@ -182,3 +164,52 @@
     program_runner.add(_prog)
     results = await program_runner.join()
     assert len(results) == n
+
+
+@mark.asyncio
+async def test_invert_share(test_preprocessing):
```

Comment:
Do you want to consider adding a negative test case where you don't pass the mixin and check for `NotImplementedError`?

Reply:
I'll add this-- good catch